launch/launch/__init__.py

```diff
@@ -75,8 +78,11 @@ def __init__(self, cmd, name, output_handler, exit_handler, env=None):
 
     def send_signal(self, signal):
         if self.transport:
+            self.task_state.signals_received.append(signal)
             self.transport.send_signal(signal)
 
     def terminate(self):
-        if self.transport:
+        if os.name != 'nt':
+            self.send_signal(signal.SIGINT)
```

Comment:
It needs to be made sure that the process has "enough" time to process the `SIGINT`. Otherwise it might be `SIGTERM`ed shortly after.

I think it should be ensured that this process actually terminates in a finite time. Independent of the fact that the launch file will ensure that when its shutting down. Not sure what a good way to achieve that is.